autofit/non_linear/abstract_search.py

```diff
@@ -445,6 +445,8 @@ class represented by model M and gives a score for their fitness.
         self.paths.unique_tag = self.unique_tag
         self.paths.restore()
 
+        analysis.modify_before_fit(model=model)

```

Comment:
Might be worth making this return a new analysis rather than modifying the existing one. Otherwise when you reuse the analysis for different searches you might get unexpected results

Reply:
Do you mean a copy of the analysis, or to simple use return analysis?